### PR TITLE
Make comments toggle by click

### PR DESCRIPTION
**Pull request by LISBON11**

### CODE REVIEW

File: common.blocks/issue/issue.js [hunk 1]
```diff
@@ -1,7 +1,9 @@
-modules.define('issue', ['i-bem-dom', 'jquery', 'button', 'functions', 'api-request'],
-    function(provide, bemDom, $, Button, Functions, request) {
+modules.define('issue', ['i-bem-dom', 'jquery', 'button', 'comments', 'functions', 'api-request'],
+    function(provide, bemDom, $, Button, Comments, Functions, request) {
 
 provide(bemDom.declBlock(this.name, {
+    isCommentsOpened: false,
```
Comment (volkov97) [inline common.blocks/issue/issue.js]: you need to make this variable private and put into this

Comment (volkov97) [reply common.blocks/issue/issue.js]: and i think you should do it in inited function

File: common.blocks/issue/issue.js [hunk 2]
```diff
@@ -36,13 +38,21 @@ provide(bemDom.declBlock(this.name, {
...
         event.preventDefault();
-        button.setMod('disabled');
 
-        request(button.params.number + '/comments')
-            .then(function(response) { return response.text(); })
-            .then(function(data) {
-                bemDom.append(footer.domElem, data);
-            });
+        if (!this.isCommentsOpened) {
+            this.isCommentsOpened = true;
```
Comment (volkov97) [inline common.blocks/issue/issue.js]: you need to do a private function called like _toggleCommentsVisibility
in that function you should check present state of this._isCommentsVisible and invert it

Comment (volkov97) [reply common.blocks/issue/issue.js]: and tou need to call this function once after this checks

File: common.blocks/issue/issue.js [hunk 3]
```diff
@@ -36,13 +38,21 @@ provide(bemDom.declBlock(this.name, {
...
+
+            request(button.params.number + '/comments')
+                .then(function(response) { return response.text(); })
+                .then(function(data) {
+                    bemDom.append(footer.domElem, data);
+                });
+        } else {
+            var comments = this.findChildBlock(Comments);
```
Comment (volkov97) [inline common.blocks/issue/issue.js]: you don't need this variable - just use
bemDom.destruct(this.findChildBlock(Comments).domElem);

File: common.blocks/issue/issue.js [hunk 4]
```diff
@@ -36,13 +38,21 @@ provide(bemDom.declBlock(this.name, {
...
 
         event.preventDefault();
-        button.setMod('disabled');
 
-        request(button.params.number + '/comments')
-            .then(function(response) { return response.text(); })
-            .then(function(data) {
-                bemDom.append(footer.domElem, data);
-            });
+        if (!this.isCommentsOpened) {
```
Comment (volkov97) [inline common.blocks/issue/issue.js]: i think it's better first to check if comments are visible, and then that they are not

**Comment by skad0** [comment]
I think that it isn't necessary to load comments every time we toggle them. 
Maybe it should be just visibility modifier? And later we should implement autoload of new comments.
@tadatuta

**Comment by tadatuta** [comment]
@skad0 👍

**Pull request merged**